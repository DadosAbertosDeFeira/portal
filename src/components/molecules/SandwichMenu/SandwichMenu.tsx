import { Button, type ButtonProps } from "atoms/Button";
import classNames from "classnames";
import React, { forwardRef } from "react";
import { BiMenu } from "react-icons/bi";

export interface SandwichMenuProps extends ButtonProps {
  isOpen: boolean;
}

export const SandwichMenu = forwardRef<HTMLButtonElement, SandwichMenuProps>(
  function SandwichMenu({ isOpen, className, ...props }, ref) {
    return (
      <Button
        {...props}
        aria-label="Menu"
        type="button"
        className={classNames(
          "transition-all duration-150",
          isOpen ? "rotate-[-90deg]" : "rotate-0",
          className
        )}
        ref={ref}
      >
        <BiMenu size={24} />
      </Button>
    );
  }
);
