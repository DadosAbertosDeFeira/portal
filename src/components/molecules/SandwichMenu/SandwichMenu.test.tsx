import { render } from "@testing-library/react";

import { SandwichMenu, type SandwichMenuProps } from "./SandwichMenu";

describe("SandwichMenu", () => {
  const makeSut = ({
    isOpen = false,
    ...props
  }: Partial<SandwichMenuProps>) => {
    return render(<SandwichMenu isOpen={isOpen} {...props} />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });

  it("should render open correctly", () => {
    const { asFragment } = makeSut({ isOpen: true });

    expect(asFragment).toMatchSnapshot();
  });
});
