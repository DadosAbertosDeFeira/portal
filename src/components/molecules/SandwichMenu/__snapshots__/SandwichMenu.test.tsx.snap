// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SandwichMenu renders correctly 1`] = `[Function]`;

exports[`SandwichMenu should render open correctly 1`] = `[Function]`;
